types/alks-node/index.d.ts

```diff
@@ -0,0 +1,125 @@
+// Type definitions for alks-node 0.9
+// Project: https://github.com/Cox-Automotive/alks-node#readme
+// Definitions by: Matt Hoang <https://github.com/matthoang08>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import { Moment } from 'moment';
+
+export interface Auth {
+    token?: string;
+    password?: string;
+}
+export interface Account {
+    userid: string;
+    server: string;
+    alksAccount: string;
+    alksRole: string;
+}
+export interface AlksAccount {
+    account: string;
+    role: string;
+    iam?: boolean;
+}
+export interface LongTermKeyData {
+    accessKey: string;
+    secretKey: string;
+    iamUserName: string;
+    iamUserArn: string;
+    alksAccount: string;
+    alksRole: string;
+}
+export interface KeyData {
+    accessKey: string;
+    secretKey: string;
+    sessionToken: string;
+    alksAccount: string;
+    alksRole: string;
+    sessionTime: string;
+    expires: Moment;
+}
+export interface AwsKey {
+    accessKey: string;
+    secretKey: string;
+    sessionToken: string;
+}
+export function getDurations(
+    account: Account,
+    auth: Auth,
+    opts: object,
+    callback: (err: Error, duration: number[]) => void
+): void;
+export function createKey(
+    account: Account,
+    auth: Auth,
```

Comment:
This parameter is named `password` and based on the examples looks like is a `string` type.

Reply:
Documentation is outdated: function definition:
https://github.com/Cox-Automotive/alks-node/blob/b2a951245421ada5ba0c856e2c167c5808c16025/lib/alks-api.js#L122

Auth object is used here if it contains token otherwise use password:
https://github.com/Cox-Automotive/alks-node/blob/b2a951245421ada5ba0c856e2c167c5808c16025/lib/alks-api.js#L52